packages/lesswrong/components/posts/PostsPage/PostsPage.tsx

```diff
@@ -325,6 +329,26 @@ const PostsPage = ({post, eagerPostComments, refetch, classes}: {
 
   const { query, params } = location;
 
+  useEffect(() => {
+    if (!query[SHARE_POPUP_QUERY_PARAM]) return;
+
+    openDialog({
+      componentName: "SharePostPopup",
+      componentProps: {
+        post,
+      },
+      noClickawayCancel: true,
+      closeOnNavigate: true,
+    });
+
+    // Remove "sharePopup" from query once the popup is open, to prevent accidentally
+    // sharing links with the popup open
+    const currentQuery = isEmpty(query) ? {} : query
+    const newQuery = {...currentQuery, [SHARE_POPUP_QUERY_PARAM]: undefined}
+    history.push({...location.location, search: `?${qs.stringify(newQuery)}`})
+
+  }, [query[SHARE_POPUP_QUERY_PARAM]]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Action not _necessarily_ needed, but just noting that I think disabling exhaustive-deps is basically always wrong and has a habit of leading to subtle bugs in the future that are a nightmare to track down.

Reply:
👍 In this case it isn't needed anymore anyway because the query param is removed after opening the popup